src/com/dotcms/rest/BaseRestPortlet.java

```diff
@@ -184,24 +187,74 @@ public Response getLayout ( @Context HttpServletRequest request, @Context HttpSe
 
 	}
 
-	private class ResponseWrapper extends HttpServletResponseWrapper {
-
-		private StringWriter writer = new StringWriter();
+	public class ResponseWrapper extends HttpServletResponseWrapper{
+		private ByteArrayOutputStream output;
+		private StringWriter writer;
+		private int contentLength;
+		private String contentType;
 
 		public ResponseWrapper(HttpServletResponse response) {
 			super(response);
+			output = new ByteArrayOutputStream();
+			writer = new StringWriter();
+		}
+
+		public ServletOutputStream getOutputStream() {
+			return new FilterServletOutputStream(output);
+		}
+
+		public byte[] getData() {
+			return output.toByteArray();
 		}
 
 		public String getResponseString() {
-			return writer.toString();
+			return UtilMethods.isSet(writer.toString()) ? writer.toString() : output.toString();
 		}
 
-		@Override
 		public PrintWriter getWriter() throws IOException {
 			PrintWriter pw = new PrintWriter(writer);
 			return pw;
 		}
 
+		public void setContentType(String type) {
+			this.contentType = type;
+			super.setContentType(type);
+		}
+
+		public String getContentType() {
+			return this.contentType;
+		}
+
+		public int getContentLength() {
```

Comment:
Where is this used? If it's required (and I assume it is) please describe how it would break if removed in a comment. If it's not required then let's get rid of the contentLength overrides all together if we can.